gr-zeromq/lib/base_impl.cc

```diff
@@ -26,8 +26,12 @@ namespace zeromq {
 base_impl::base_impl(int type,
                      size_t itemsize,
                      size_t vlen,
+                     char* address,
                      int timeout,
                      bool pass_tags,
+                     int hwm,
+                     bool sink,
```

Comment:
As an implementation decision, this is one way to do it, but I generally view switching behavior based on a state value as a failure to abstract. Given that you are trying to support multiple ZMQ API versions (with incompatible argument types), this is possibly the lesser of two evils, but it is still evil.